### PR TITLE
GSYE-466: Add asset_energy_requirements_kWh to SCMKPIState

### PR DESCRIPTION
**Pull request by mrsaemir**

### CODE REVIEW

File: gsy_framework/sim_results/scm/kpi.py [hunk 1]
```diff
@@ -78,6 +79,12 @@ def update(self, energy_demanded_wh: float, energy_produced_wh: float,
         self.total_gsy_e_cost_excl_revenue += gsy_e_cost_excl_revenue
         self.total_fit_revenue += fit_revenue
 
+        for asset_uuid, asset_energy_requirement in asset_energy_requirements_kWh.items():
+            if asset_uuid in self.total_asset_energy_requirements_kWh:
+                self.total_asset_energy_requirements_kWh[asset_uuid] += asset_energy_requirement
+            else:
+                self.total_asset_energy_requirements_kWh[asset_uuid] = asset_energy_requirement
+
```
Comment (hannesdiedrich) [inline gsy_framework/sim_results/scm/kpi.py:87]: NIT, we have a `add_or_create_key` in gsy_framework.utils for this.

Comment (mrsaemir) [reply gsy_framework/sim_results/scm/kpi.py]: Fixed. Thanks.

Comment (hannesdiedrich) [reply gsy_framework/sim_results/scm/kpi.py]: wow, quick!

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Pull request merged**